pkg/prometheus_ingester/prometheus_ingester_test.go

```diff
@@ -718,3 +720,65 @@ func Test_processHistogramIncrease(t *testing.T) {
 		assert.ElementsMatchf(t, testCase.expectedEvents, generatedEvents, "expected events:\n%s\n\nresult:\n%s", testCase.expectedEvents, generatedEvents)
 	}
 }
+
+func TestPrometheusIngesterHttpHeader_UnmarshalYAML(t *testing.T) {
+	const envName = "ENVNAME"
+	const envValue = "envValue"
+
+	if err := os.Setenv(envName, envValue); err != nil {
+		t.Fatal(err)
+	}
+
+	tests := []struct {
+		name          string
+		goldenfile    string
```

Comment:
I would prefer to avoid using goldenfiles and use yaml directly within test defintion

But if you have some strong reason to do it this way, feel free to keep it as it is now. :)

Reply:
seemed prettier for me to have "yamls" in separated file, but I don't insist on it...

fixed d48902d4ad77ea48e830ce5ce24c2a6aa5fce662